hudi-client/src/main/java/org/apache/hudi/client/AbstractHoodieWriteClient.java

```diff
@@ -117,7 +118,17 @@ public boolean commitStats(String instantTime, List<HoodieWriteStat> stats, Opti
     if (extraMetadata.isPresent()) {
       extraMetadata.get().forEach(metadata::addMetadata);
     }
-    metadata.addMetadata(HoodieCommitMetadata.SCHEMA_KEY, config.getSchema());
+    String schema = config.getSchema();
+    if (config.updatePartialFields()) {
+      try {
+        TableSchemaResolver resolver = new TableSchemaResolver(table.getMetaClient());
```

Comment:
> Do you need to create resolver again? Does config.getLastSchema() work here?

I tested it E2E and found that would not get the lastSchema from config since the config object are different

Reply:
We can get the last Schema from the commit metadata, right?